src/Common/parseGlobs.cpp

```diff
@@ -18,21 +21,21 @@ namespace DB
  */
 std::string makeRegexpPatternFromGlobs(const std::string & initial_str_with_globs)
 {
-    std::ostringstream oss_for_escaping;
-    oss_for_escaping.exceptions(std::ios::failbit);
+    /// FIXME make it better
+    WriteBufferFromOwnString buf_for_escaping;
     /// Escaping only characters that not used in glob syntax
     for (const auto & letter : initial_str_with_globs)
     {
-        if ((letter == '[') || (letter == ']') || (letter == '|') || (letter == '+') || (letter == '-') || (letter == '(') || (letter == ')'))
-            oss_for_escaping << '\\';
-        oss_for_escaping << letter;
+        if ((letter == '[') || (letter == ']') || (letter == '|') || (letter == '+') || (letter == '-') || (letter == '(') || (letter == ')') || (letter == '\\'))
```

Comment:
This change is suspicious a little bit, but seem to be logical.

Reply:
It looks like a typical mistake with escaping: if we have something like `qwe[rty` in `initial_str_with_globs`, then we replace it with `qwe\[rty` (as expected), but if we have `qwe\[rty` in `initial_str_with_globs`, it's replaced with `qwe\\[rty`, which seems to be incorrect. Backslash should be escaped too, so `qwe\[rty` will be replaced with `qwe\\\[rty`. 
Probably we should recheck all code related to globs, it all looks suspicious, as for me.